tests/AbstractTestCase.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * ua-parser
+ *
+ * Copyright (c) 2011-2012 Dave Olsen, http://dmolsen.com
+ *
+ * Released under the MIT license
+ */
+namespace UAParser\Test;
+
+use PHPUnit\Framework\TestCase;
+
+abstract class AbstractTestCase extends TestCase
+{
+    /**
+     * Polyfill to allow exception testing across PHPUnit 4 to PHPUnit 7.
+     */
+    public function fcExpectException($exception, $message = null)
```

Comment:
Can we just polyfill setExpectedException()?

Reply:
Maybe with a trait. I will check.

Reply:
Ah no, traits are available as of PHP 5.4. Uaparser tests on 5.3 also.

Reply:
So on 2nd read we do not need a trait, just provide the method in a base class.

Reply:
Yup, that should work

Reply:
The Namespaced version for PHPUnit is available in the versions `^4.8.35 | ^5.4.3 | ^6.0.0 | ^7.0.0`.